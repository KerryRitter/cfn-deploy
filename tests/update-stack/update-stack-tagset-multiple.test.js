const mockCreateStack = require('../mocks/updateStack');
const lib = require('../..');
const stringTags = require('../tags/string-tags.json');


describe('cfn-deploy', () => {
  beforeAll(() => mockCreateStack.beforeAll());

  it('should successfully update existing stack', (done) => {
    let tagsArray = [
      './tests/tags/array-tags.json',
      './tests/tags/object-tags.json',
      './tests/tags/object-cp-tags.json',
      {
        DummyTag: 'dummy-tag',
        OverwritableTag: 'from-object-tags',
        TagSource: 'plain-object',
      },
    ];
    tagsArray = tagsArray.concat(stringTags);
    const events = lib({
      region: 'us-east-1',
      stackName: 'existing-stack',
      template: './tests/templates/simple-template.yaml',
      tags: tagsArray,
    });
    events.on('ERROR', err => done(err));
    events.on('COMPLETE', (data) => {
      expect(data).toMatchSnapshot();
    });
    events.on('FINALLY', done);
  });

  afterAll(() => mockCreateStack.afterAll());
});
