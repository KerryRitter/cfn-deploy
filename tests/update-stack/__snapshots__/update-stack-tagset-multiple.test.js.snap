// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cfn-deploy should successfully update existing stack 1`] = `
Object {
  "Capabilities": Array [],
  "ChangeSetName": "cfn-deploy",
  "ChangeSetType": "UPDATE",
  "Description": "Created with cfn-deploy",
  "Parameters": Array [],
  "StackName": "existing-stack",
  "Tags": Array [
    Object {
      "Key": "DummyTag",
      "Value": "dummy-tag",
    },
    Object {
      "Key": "OverwritableTag",
      "Value": "from-array-tags-file",
    },
    Object {
      "Key": "TagSource",
      "Value": "array-params.json",
    },
    Object {
      "Key": "DummyTag",
      "Value": "dummy-tag",
    },
    Object {
      "Key": "OverwritableTag",
      "Value": "from-object-tags-file",
    },
    Object {
      "Key": "TagSource",
      "Value": "object-tags.json",
    },
    Object {
      "Key": "DummyTag",
      "Value": "dummy-tag",
    },
    Object {
      "Key": "OverwritableTag",
      "Value": "from-object-tags",
    },
    Object {
      "Key": "TagSource",
      "Value": "plain-object",
    },
    Object {
      "Key": "DummyTag",
      "Value": "dummy-tag",
    },
    Object {
      "Key": "OverwritableTag",
      "Value": "from-string-tags-file",
    },
    Object {
      "Key": "TagSource",
      "Value": "string-tags.json",
    },
  ],
  "TemplateBody": "Resources:
  S3Bucket:
    Type: AWS::S3::Bucket
",
}
`;

exports[`cfn-deploy should successfully update existing stack 2`] = `
Object {
  "StackStatus": "UPDATE_COMPLETE",
}
`;
